src/routes/prepare_game.py

```diff
@@ -1,104 +1,64 @@
-from fastapi import APIRouter, Request, BackgroundTasks
-from fastapi.responses import JSONResponse
-from src.game_managment import GameSession
-from pydantic import Field, BaseModel
-from pyndemic.game import Game
-from pyndemic.formatter import BaseFormatter
-from datetime import datetime
 import logging
-from typing import Dict
+
+from fastapi import APIRouter, Request
+from fastapi.responses import JSONResponse
+from pyndemic.controller import GameController
+from pydantic import BaseModel
+
+from src.game_managment import find_game, GameIsMissingError
 
 logger = logging.getLogger(__name__)
 
 
 router = APIRouter()
 
 
-class GameConfig(BaseModel):
-    owner_name: str
-    # difficult: int = Field(..., ge=1, le=4,
-    #                        description='Difficult of the game 1 to 4 there 1 is very easy and 4 is very hard')
-    # player_count: int = Field(..., ge=1, le=6,
-    #                           description='Count playeer for new game')
-    difficult: str
-    player_count: str
+class GameOptions(BaseModel):
+    # It's temporary arguments, while we don't know which use
+    difficult: int = None
+    player_count: int = None
 
 
-class GameDescription(BaseModel):
-    name: str
-    owner_name: str
-    difficult: int
-    players: Dict[str, str]
-    player_count: int
-    created: str
+class JoinGameOptions(BaseModel):
+    game_id: str
 
 
-@router.get(
+@router.post(
     '/create_new_game',
-    description='Create instance of the Game',
-    response_model=GameDescription
+    description='Create instance of the Game'
 )
-async def create_new_game(
-    request: Request,
-    game_config: GameConfig,
-    background_tasks: BackgroundTasks,
-):
-    game = Game()
-    current_player = request.state.player
-    background_tasks.add_task(setup_game, game)
-    gameSession = GameSession(owner_name=game_config.owner_name,
-                              owner_user_id=current_player,
-                              difficult=game_config.difficult,
-                              player_count=game_config.player_count,
-                              created=datetime.utcnow().isoformat(),
-                              game=game)
-
-    request.app.state.games.update({current_player: gameSession})
-
-    response = {'name': current_player,
-                'owner': current_player,
-                'difficult': gameSession.difficult,
-                'players': gameSession.players,
-                'player_count': f'1 of {gameSession.player_count}',
-                'created': gameSession.created}
-    return response
+async def create_new_game(request: Request, game_options: GameOptions):
+
+    # TODO bind request.state.session with controller or keep game_id in session
 
+    controller = GameController()
+    game_id = controller._ctx
```

Comment:
Did you mean this ID?
```suggestion
    game_id = controller._ctx['id']
```